### PR TITLE
#220 | Crash Fixes

### LINKED ISSUE #220: Game Crashes starting over after winning

### LINKED ISSUE #221: Win Screen and Credit Music Overlap

### LINKED ISSUE #222: Combat - Give player a key after killing the boss

### PR DESCRIPTION
**Pull request by demwilson**
* Fixed an error being thrown when returning to title from a victory.
* Fixed an issue where music from Win and Credits would play at the same time.
* Player receives a key after defeating the boss.
* Reworked how boss monster is added to combat.
* Fixed the search by id for enemies.
* Readded the background color to combat instructions.

Closes #220 
Closes #221 
Closes #222

### CODE REVIEW

File: game/Items.gd [hunk 1]
```diff
@@ -95,8 +96,11 @@ func generate_items(tier_level, count=DEFAULT_COUNT):
 			item_list.append(item)
 	return item_list
 
+func generate_boss_key():
+	var item = get_item_by_id(ItemList.BOSS_KEY)
+	return LootItem.new(LootType.GEAR, item)
```
Comment (beserkzombie) [inline game/Items.gd:103]: Should also handle setting the flag of _floor_key to true after generating the key

Comment (beserkzombie) [reply game/Items.gd:103]: And remove where I set it in overworld after calling combat.

File: Global.gd [hunk 2]
```diff
@@ -201,51 +201,54 @@ func _ready():
 		Item.new(36, "Mono Blade", Item.ItemTier.LEVEL_THREE, Item.ItemType.MOVE, "You swing the mono balde.", Moves.MoveList.MELEE_T3),
 		Item.new(37, "Psychic Immolation", Item.ItemTier.LEVEL_THREE, Item.ItemType.MOVE, "Let them live in the flames!", Moves.MoveList.PSY_T3),
 		Item.new(38, "Healing Nanites", Item.ItemTier.LEVEL_THREE, Item.ItemType.MOVE, "Healing yourself is below you.", Moves.MoveList.HEAL_T3),
+		
+		# Unique Items
+		Item.new(39, "Boss Key", Item.ItemTier.LEVEL_ZERO, Item.ItemType.KEY, "This key probably opens something!", null),
```
Comment (beserkzombie) [inline Global.gd:206]: Could we make these giant object arrays into a Json file we auto load?

Comment (demwilson) [reply Global.gd:206]: Not in time for the submission.

I created a ticket for it: #224

File: game/Items.gd (hunk 1 above)
Comment (demwilson) [reply game/Items.gd:103]: Done with new push.

Comment (beserkzombie) [reply game/Items.gd:103]: I don’t see a change

Comment (demwilson) [reply game/Items.gd:103]: I'm doing the work in a different location:
image

**Review by beserkzombie — APPROVED** [review]
K, looks good.

**Pull request merged**